src/MatrixClient.ts

```diff
@@ -1257,60 +1259,53 @@ export class MatrixClient extends EventEmitter {
             if (body && Buffer.isBuffer(body)) LogService.debug("MatrixLiteClient (REQ-" + requestId + ")", "body = <Buffer>");
         }
 
-        const params: { [k: string]: any } = {
-            uri: url,
+        const params: OptionsOfDefaultResponseBody = {
+            url: url,
             method: method,
-            qs: qs,
-            // If this is undefined, then a string will be returned. If it's null, a Buffer will be returned.
-            encoding: noEncoding === false ? undefined : null,
-            userQuerystring: true,
-            qsStringifyOptions: {
-                options: {arrayFormat: 'repeat'},
-            },
+            searchParams: qs,
+            // @ts-ignore
+            responseType: noEncoding === false ? "text" : "buffer",
             timeout: timeout,
             headers: headers,
+            allowGetBody: true,
         };
 
         if (Buffer.isBuffer(body)) {
             params.headers["Content-Type"] = contentType;
             params.body = body;
+        } else if (body === null) {
```

Comment:
Thanks but....the fix is merged in matrix-js-bot-sdk develop and it looks like wrapping the thing in

```
if (body) {
    // stuff
}
```

https://github.com/turt2live/matrix-js-bot-sdk/blob/develop/src/MatrixClient.ts#L1274-L1282